sense-app/src/main/kotlin/com/exactpro/th2/sense/app/source/processor/SenseProcessorFactory.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023-2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.sense.app.source.processor
+
+import com.exactpro.th2.dataprovider.lw.grpc.DataProviderService
+import com.exactpro.th2.processor.api.IProcessor
+import com.exactpro.th2.processor.api.IProcessorFactory
+import com.exactpro.th2.processor.api.IProcessorSettings
+import com.exactpro.th2.processor.api.ProcessorContext
+import com.exactpro.th2.sense.api.*
+import com.exactpro.th2.sense.app.cfg.*
+import com.exactpro.th2.sense.app.notifier.event.GrafanaEventNotifier
+import com.exactpro.th2.sense.app.processor.event.EventProcessorListener
+import com.exactpro.th2.sense.app.processor.impl.*
+import com.exactpro.th2.sense.app.statistic.impl.AggregatedEventStatistic
+import com.exactpro.th2.sense.app.statistic.impl.BucketEventStatistic
+import com.exactpro.th2.sense.app.statistic.impl.GrafanaEventStatistic
+import com.exactpro.th2.sense.app.statistic.impl.NotificationEventStatistic
+import com.exactpro.th2.sense.app.notifier.grpc.SenseService
+import com.exactpro.th2.sense.app.notifier.http.SenseHttpServer
+import com.exactpro.th2.sense.app.source.Source
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.jsontype.NamedType
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.google.auto.service.AutoService
+import io.grpc.BindableService
+import mu.KotlinLogging
+import java.time.Instant
+import java.util.Deque
+import java.util.concurrent.ConcurrentLinkedDeque
+import com.exactpro.th2.common.event.Event as CommonEvent
+
+@AutoService(IProcessorFactory::class)
+class SenseProcessorFactory : IProcessorFactory {
+    private val processorProvider = ServiceLoaderProcessorProvider.create<EventProcessor, EventProcessorFactory<*>>()
+    private val resources: Deque<() -> Unit> = ConcurrentLinkedDeque()
```

Comment:
It is redundant queue, because the factory doesn't control resources. Processor should close all resources on Close method call

Reply:
Please remove unused variable